subprojects/files/src/main/java/org/gradle/api/internal/file/archive/impl/FileZipInput.java

```diff
@@ -57,7 +57,7 @@ private static boolean isZipFileSafeToUse() {
 
     private static int getJavaMajorVersion() {
         String versionString = System.getProperty("java.version");
-        String[] versionParts = versionString.split("\\.");
+        String[] versionParts = versionString.split("\\.|-ea");
```

Comment:
This should use `java.specification.version` instead.

Reply:
See: https://openjdk.java.net/jeps/223#System-properties

Reply:
Thanks!

Reply:
Note that before Java 9 the format of specification was `1.8`, but you can still easily parse that as a number, or reject anything starting with `1.`.